app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -157,7 +157,7 @@ public class LoyaltyCardEditActivity extends CatimaAppCompatActivity {
     String cardId;
     String barcodeId;
     String barcodeType;
-    String addGroup;
+    int addGroup;
```

Comment:
Because `addGroup` can be null, this should be `Integer` (the nullable version of `int`) instead.

```suggestion
    Integer addGroup;
```